src/main/java/org/graylog/integrations/dataadapters/GreyNoiseCommunityIpLookupAdapter.java

```diff
@@ -125,29 +131,45 @@ protected LookupResult doGet(Object ipAddress) {
         return result;
     }
 
-    private Request createRequest(Object ipAddress) {
-        String ipString = ipAddress == null? "" : ipAddress.toString();
-        if(ipString.trim().isEmpty()){
-            String error = String.format("[%s] requires an IP address to perform Lookup",ADAPTER_NAME);
-            throw new IllegalArgumentException(error);
-        }
+    @VisibleForTesting
+    Request createRequest(Object ipAddress) {
+        String ipString = getValidIpString(ipAddress);
 
-        String userAgent = String.format(USER_AGENT, Version.CURRENT_CLASSPATH);
         String apiToken = encryptedValueService.decrypt(config.apiToken());
-        if(apiToken == null || apiToken.trim().isEmpty()){
-            String error = String.format("[%s] requires a non-null API Token",ADAPTER_NAME);
+        if (apiToken == null || apiToken.trim().isEmpty()) {
+            String error = String.format("[%s] requires a non-null API Token", ADAPTER_NAME);
             throw new IllegalArgumentException(error);
         }
 
         return new Request.Builder()
                 .url(String.join("/", GREYNOISE_COMMUNITY_ENDPOINT, ipString))
                 .method(METHOD, null)
                 .addHeader("Accept", ACCEPT_TYPE)
-                .addHeader("User-Agent", userAgent)
+                .addHeader("User-Agent", USER_AGENT)
                 .addHeader("key", apiToken)
                 .build();
     }
 
+    private String getValidIpString(Object ipAddress) {
+        String ipString = ipAddress == null ? "" : ipAddress.toString();
+        if (ipString.trim().isEmpty()) {
+            String error = String.format("[%s] requires an IP address to perform Lookup", ADAPTER_NAME);
+            throw new IllegalArgumentException(error);
+        }
+
+        if (!InetAddressValidator.getInstance().isValidInet4Address(ipString)) {
+            String error = String.format("[%s] is not a valid IPv4 Address", ipString);
```

Comment:
Should we also do a check to see if the IP is IPv6 and not throw an error if it is IPv6, but maybe just print out a debug log saying that no GreyNoise lookup will be attempted since the IP is IPv6? Throwing an error for an IPv6 address seems wrong since it technically isn't an invalid IP, it's just not a supported IP.

Reply:
Yes, I was wondering the same for a reserved IP address.  As is, the result (in the test screen) is very unhelpful:
```
{
  "single_value": null,
  "multi_value": null,
  "string_list_value": null,
  "has_error": true,
  "ttl": 5000
}```

I am not sure that we need to check whether it is IPv6, since we really only care that it is a valid IPv4--unless we want to be extra helpful, which may not be a bad idea?

Reply:
I recommend checking for IPv6 so that we can safely ignore them and not throw an error since it is perfectly valid to have an IPv6 address, we just wont send it to GreyNoise. On the off chance a user overrwrites an IP address field with an invalid value or has them sent in/parsed incorrectly I'd think we'd want different behavior than if an IPv6 address came in. And I agree, it should work the same for non-routable IPs.

Reply:
Thanks for the feeback @kingzacko1 .  I have addressed your points.  I created a PR ([#11873](https://github.com/Graylog2/graylog2-server/pull/11873)) for the reserved IP checker.